docs/guide/nts.md

```diff
@@ -0,0 +1,109 @@
+# Network Time Security
+
+Network Time Security (NTS) is an extension to the NTP protocol aimed at securing the communication between NTP clients and servers. It guarantees that the severs responses really come from the server. Additionally, it contains countermeasures to ensure that NTP traffic cannot be used to track a specific NTP client across multiple networks. Note that it does not hide the contents of the request or response, so a third party can tell that time is exchanged and the specific time associated with the packets at the server.
+
+## Configuring an NTS server as time source
+
+Configuring a public NTS server, for example those provided by [netnod](https://www.netnod.se/nts/network-time-security), is fairly straightforward:
+```toml
+[[peer]]
+mode = "nts"
+address = "nts.netnod.se"
+```
+
+The above assumes that the certificate used by the server is accepted through the root certificates of the system. If this is not the case, the root certificate used in the certificate chain of the server needs to be separately provided to ntpd-rs. This can be done through the `certificate-authority` option:
+```toml
+[[peer]]
+mode = "nts"
+address = "my.private.server"
+certificate-authority = "/path/to/server/root/certificate.pem"
+```
+The root certificate file provided in the above example configuration should be a pem encoded list of root certificates that should be accepted for the server in addition to those in the system certificate store.
+
+Currently, ntpd-rs does not support using self-signed certificates. If non public certificates are needed, the best option is to create a private CA locally and use that to sign the server's certificate. Instructions for that are included below in the instructions for configuring ntpd-rs as an NTS server.
+
+## Configuring ntpd-rs as an NTS server
+
+To configure ntpd-rs as a server, we need to configure three things. First, ntpd-rs needs to be configured to work as a normal NTP server. Second, we need to enable the server components that allow key exchange with clients. Finally, we need to configure storage for internal keys to ensure clients don't need to reconnect on a reboot of the server.
+
+The process of setting up an NTS server is significantly harder than configuring an NTS time source. We assume here some experience with [setting up a non-NTS NTP server](TODO), and will assume you already have that running on the target machine.
+
+Furthermore, NTPD-rs currently does not support running an NTS server on a server without a domain name through which its clients reach it. This can however be a local domain name, such as one configured through `/etc/hosts`.
```

Comment:
what does this mean? can we have an example here of what is/is not possible?

Reply:
done